cinema/views.py

```diff
@@ -101,8 +104,27 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        url_path="upload-image",
+        permission_classes=[IsAdminUser]
```

Comment:
You have already defined IsAdminOrIfAuthenticatedReadOnly permission class, so this is redundant here